internal/cmd: more consolidation and use of the operator.Configuration type

joelanford opened the pull request:
**Description of the change:**

Internal refactoring to make more re-use of existing code.

- Use `operator.Configuration` in scorecard.
- Remove duplicative scorecard-specific client config helpers.
- Move `operator.Configuration` load call for the `run` command to the root of the run command so it can be reused by both `packagemanifests` and `bundle`.

**Motivation for the change:**
Reduce code duplication

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] ~Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))~
- [ ] ~Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)~

jberkhahn commented:
So, not to look a gift horse in the mouth or anything, but if it is a goal to eventually have all our tests in ginkgo/gomega (which it is, i think?), why are we still adding raw gotest tests?

joelanford commented:
@jberkhahn These tests existed already in for the scorecard code that was removed. So I moved those tests over and refactored them a little to work for `config.Load`.

camilamacedo86 commented on internal/cmd/operator-sdk/scorecard/cmd.go:
```diff
@@ -28,50 +28,55 @@ import (
+			if err := c.validate(args); err != nil {
+				return err
+			}
+			return c.cfg.Load()
```
👍

camilamacedo86 commented on internal/cmd/operator-sdk/scorecard/cmd.go:
```diff
@@ -152,19 +157,19 @@ func (c *scorecardCmd) run() (err error) {
-
 		// Only get the client if running tests.
-		if runner.Client, err = scorecard.GetKubeClient(c.kubeconfig); err != nil {
+		kubeClient, err := kubernetes.NewForConfig(c.cfg.RESTConfig)
```
👍

camilamacedo86 reviewed (approved):
It shows great for me 👍 
/lgtm

camilamacedo86 commented:
Hi @joelanford, 

Let's rebase it with the master and get it merged?

camilamacedo86 reviewed (commented):
@joelanford 

We will lose this cleanup. Is it no longer required?
